transcom/mymove
# MB-3550 Show backup contact info in Move details

## Description

In order to display backup contact info on the Move details page (as
seen by a TOO), the GHC API had to be modified to include that data.

Note that the Service Member model refers to Backup Contacts as a
`has_many` relationship, but in practice, a service member can only
add a single backup contact during onboarding. For that reason, I
named the key in the payload `backup_contact`.

For now, I only included the name, email, and phone attributes in the
swagger definition. If it's expected that a TOO should be able to
make changes to the backup contact, then we'll need to add the `id`
and `eTag`.

Because the backup contact `phone` is optional, I made it appear last
in the Customer info component.

## Reviewer Notes

Steps to test:
1. Run the setup commands in the section below
2. Visit http://officelocal:3000/devlocal-auth/login
2. Click the `Login` button next to `too_role@office.mil (PPM office)`
3. Click on the last `MoveOrderID` in the list
4. Scroll all the way down to the Customer Info section

Expected Result: you should see the Backup Contact name, email, and phone populated. If you don't, try another move. Most of the generated moves include backup contact info.

Other scenarios to test:

- When no backup contact exists. This should be the case for the first move in the list. If not, try other moves until you find one that doesn't have backup contact info and make sure you see blank fields, and that you don't get any errors or unexpected `undefined` values in the Move Details page.

- When backup contact exists, but without a phone. For this, you'll need to create a new move manually, and when you get to the backup contact info section, make sure to leave the phone blank. Once you submit the move, follow steps 2-4 above to view the move you just created. You should see your backup contact name and email, and a blank phone.

See screenshots below for reference.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
 make db_dev_e2e_populate
make server_run
make office_client_run (in a separate terminal tab)
```

## Code Review Verification Steps

 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3550) for this change

## Screenshots

When all backup contact fields are present:

<img width="1032" alt="Screen Shot 2020-09-15 at 3 30 13 PM" src="https://user-images.githubusercontent.com/811150/93255795-66b4b600-f768-11ea-839b-2fe37ba1c913.png">

When the phone is not present (it is optional):
<img width="1000" alt="Screen Shot 2020-09-15 at 3 35 47 PM" src="https://user-images.githubusercontent.com/811150/93256304-27d33000-f769-11ea-979d-2f993a6571aa.png">


When a backup contact doesn't exist. This should be rare because it's required during onboarding.

<img width="995" alt="Screen Shot 2020-09-15 at 3 32 21 PM" src="https://user-images.githubusercontent.com/811150/93255987-abd8e800-f768-11ea-8e8b-6342142f9ed1.png">
